app/org/sagebionetworks/bridge/services/UserProfileServiceImpl.java

```diff
@@ -58,6 +84,67 @@ public User updateProfile(User user, UserProfile profile) {
         return user;
     }
     
+    @Override
+    public void sendStudyParticipantRoster(Study study) {
+        checkNotNull(study, Validate.CANNOT_BE_NULL, "study");
+
+        executorService.submit(new ParticipantRosterGenerator(study));
+    }
+    
+    private class ParticipantRosterGenerator implements Runnable {
+        private final Study study;
+        
+        public ParticipantRosterGenerator(Study study) {
+            this.study = study;
+        }
+        
+        @Override
+        public void run() {
+            try {
+                List<StudyParticipant> participants = Lists.newArrayList();
+
+                // We must iterate over every account in the application, not just the study directory, in case 
+                // a user signed up with a different study. There is no way to search/query for specific custom 
+                // data, we must iterate over all accounts. Will this exceed 30 seconds eventually? It could.
+                // One solution would be to do this in a worker thread and email the results to the researcher.
+                String consentKey = study.getIdentifier() + BridgeConstants.CUSTOM_DATA_CONSENT_SIGNATURE_SUFFIX; 
+                Application app = StormpathFactory.getStormpathApplication(StormpathFactory.getStormpathClient());
+                
+                StormpathAccountIterator iterator = new StormpathAccountIterator(app);
+                for (List<Account> page : iterator) {
+                    for (Account account : page) {
+                        if (account.getCustomData().containsKey(consentKey)) {
+                            UserProfile profile = profileFromAccount(account);
+                            StudyParticipant p = new StudyParticipant();
+                            p.setFirstName(profile.getFirstName());
+                            p.setLastName(profile.getLastName());
+                            p.setEmail(profile.getEmail());
+                            p.setPhone(profile.getPhone());
+                            participants.add(p);
+                        }
+                    }
+                }
+                Collections.sort(participants, STUDY_PARTICIPANT_COMPARATOR);
+                sendMailService.sendStudyParticipantsRoster(study, participants);
```

Comment:
In our tests, we should have our mock iterator return the participants out of order, so we can exercise our comparator. Also, we should mock out the sendMailService so we can validate that we're properly filtering and sorting our participant list.